tsdb/meta.go

```diff
@@ -1360,11 +1349,16 @@ func (s *Series) UnmarshalBinary(buf []byte) error {
 
 // InitializeShards initializes the list of shards.
 func (s *Series) InitializeShards() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
This doesn't need to be deferred, and can just be called directly after `s.shardIDs` is assigned.